src/NLog/Config/LoggingConfigurationParser.cs

```diff
@@ -213,6 +216,30 @@ protected void LoadConfig(ILoggingConfigurationElement nlogConfig, string basePa
             }
         }
 
+        private static string ExpandFilePathVariables(string internalLogFile)
+        {
+            try
+            {
+#if !SILVERLIGHT
+                const string currentDir = "${currentdir}";
+                int currentDirPos = internalLogFile.IndexOf(currentDir, StringComparison.OrdinalIgnoreCase);
+                if (currentDirPos >= 0)
+                    internalLogFile = internalLogFile.Replace(internalLogFile.Substring(currentDirPos, currentDir.Length), System.IO.Directory.GetCurrentDirectory() + System.IO.Path.DirectorySeparatorChar.ToString());
+                const string baseDir = "${basedir}";
+                int baseDirPos = internalLogFile.IndexOf(baseDir, StringComparison.OrdinalIgnoreCase);
+                if (baseDirPos >= 0)
+                    internalLogFile = internalLogFile.Replace(internalLogFile.Substring(baseDirPos, baseDir.Length), LogFactory.CurrentAppDomain.BaseDirectory + System.IO.Path.DirectorySeparatorChar.ToString());
```

Comment:
could you please create a method too reduce the code duplication here? (the 3 lines)

Reply:
Done